pallets/runtime/src/bridge.rs

```diff
@@ -29,68 +27,70 @@ pub trait Trait: multisig::Trait {
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub enum IssueRecipient<AccountId> {
-    Account(AccountId),
-    Identity(IdentityId),
+pub enum BridgeTxStatus {
+    /// No such tx in the system.
+    Absent,
+    /// Tx missing cdd or bridge module frozen.
+    /// u8 represents number of times the module tried processing this tx.
+    /// It will be retried automatically. Anyone can manually retry these.
+    Pending(u8),
+    /// Tx frozen by admin. It will not be retried automatically.
+    Frozen,
+    /// Tx pending first execution. These can not be manually triggered by normal accounts.
+    Timelocked,
+    /// Tx has been credited.
+    Handled,
 }
 
-/// Converts an `IssueRecipient` to the identity and the account of the recipient. The returned
-/// error covers the case when there is a bug in the code and the account ID does not encode as
-/// bytes.
-fn identity_and_account<'a, T: 'a + Trait>(
-    recipient: &'a IssueRecipient<T::AccountId>,
-) -> StdResult<(Option<IdentityId>, Option<&'a T::AccountId>), &'static str> {
-    Ok(match recipient {
-        IssueRecipient::Account(account_id) => {
-            let to_key = AccountKey::try_from(account_id.encode())?;
-            (
-                <identity::Module<T>>::get_identity(&to_key),
-                Some(account_id),
-            )
-        }
-        IssueRecipient::Identity(did) => (Some(*did), None),
-    })
+impl Default for BridgeTxStatus {
+    fn default() -> Self {
+        BridgeTxStatus::Absent
+    }
 }
 
-/// A unique lock-and-mint bridge transaction containing Ethereum transaction data and a bridge nonce.
-#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct BridgeTx<AccountId, Balance> {
-    /// Bridge signer runtime nonce.
-    pub nonce: u64,
-    /// Recipient of POLY on Polymesh: the deposit address or identity.
-    pub recipient: IssueRecipient<AccountId>,
-    /// Amount of tokens locked on Ethereum.
+/// A unique lock-and-mint bridge transaction
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Default)]
+pub struct BridgeTx<Account, Balance> {
+    /// A single tx hash can have multiple locks. This nonce differentiates between them.
+    pub nonce: u16,
+    /// Recipient of POLYX on Polymesh: the deposit address or identity.
+    pub recipient: Account,
+    /// Amount of POLYX tokens to credit.
     pub amount: Balance,
     /// Ethereum token lock transaction hash.
     pub tx_hash: H256,
 }
 
-/// A transaction that is pending a valid identity CDD.
-#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct PendingTx<AccountId, Balance> {
-    /// The identity on which the CDD is pending.
-    pub did: IdentityId,
-    /// The pending transaction.
-    pub bridge_tx: BridgeTx<AccountId, Balance>,
+/// A unique lock-and-mint bridge transaction identifier
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Default)]
```

Comment:
I am treating nonce as nth lock in the tx hash.
The first lock of a tx hash will always get nonce 1. This breaks the requirement of bridge signers to sync up on nonces.

@vkandy is that how you are treating nonce on the relayer?

Reply:
Relayer is sending account nonce. So it's only unique per signer. But looks like it doesn't need such stricter requirement, correct? I am able to submit multiple bridge txs with same nonce.

Reply:
Hmm - I thought this was the counter in the PolyLocked contract - i.e. noOfeventsEmitted  in:
`emit PolyLocked(noOfeventsEmitted, _holder, _meshAddress, _senderBalance);`

This would then be unique per lock event.

Reply:
I completely forgot about that counter. In that case, we can actually even get away with not storing the eth tx hash at all. I am marking this PR as WIP to update the structure to use only the nonce as the identifier. Would you like to still keep the tx_hash around in tx_details or just get rid of it?